go/pkg/reader/reader.go

```diff
@@ -148,8 +148,16 @@ type compressedSeeker struct {
 	fs   ReadSeeker
 	encd *zstd.Encoder
 	// This keeps the compressed data
-	buf    *syncedBuffer
-	closed bool
+	buf *syncedBuffer
+	err error
+}
+
+var encoderInit sync.Once
+var encoders sync.Pool
+
+func newEncoder() interface{} {
+	e, _ := zstd.NewWriter(nil)
```

Comment:
You might want to use zstd.WithEncoderConcurrency(1), to avoid creating more goroutines than are necessary.

Reply:
Ah thanks!